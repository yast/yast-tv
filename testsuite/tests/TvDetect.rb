# encoding: utf-8

module Yast
  class TvDetectClient < Client
    def main
      # testedfiles: Tv.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "Tv"

      @READ_nil = { "probe" => { "tv" => nil, "dvb" => nil } }

      @READ_empty = { "probe" => { "tv" => [], "dvb" => [] } }

      @READ_det = {
        "probe" => {
          "tv"  => [
            {
              "sub_device" => "jmeno prvni",
              "unique_key" => "klic",
              "drivers"    => [{ "modules" => [] }]
            },
            {
              "device"     => "jmeno druhe",
              "unique_key" => "123xyz",
              "drivers"    => [{ "modules" => [["bttv"]] }]
            }
          ],
          "dvb" => []
        }
      }


      # Read() returns nil and invokes Report::Warning
      #     TEST (``(Tv::Detect()), [ READ_nil, $[], $[] ], nil);

      # Read() returns an empty list
      TEST(lambda { Tv.Detect }, [@READ_empty, {}, {}], nil)
      DUMP(Tv.detected_cards)

      TEST(lambda { Tv.Detect }, [@READ_det, {}, {}], nil)
      DUMP(Tv.detected_cards)

      nil
    end
  end
end

Yast::TvDetectClient.new.main
