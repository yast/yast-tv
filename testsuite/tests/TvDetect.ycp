{
    // testedfiles: Tv.ycp

    include "testsuite.ycp";

    map READ = $[
	"target" : $[ "size": -1, "tmpdir" : "/tmp", ]
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

    map READ_nil =
    $[ "probe":
		$[ "tv": nil,
		   "dvb": nil
		]
	];

    map READ_empty =
    $[ "probe":
		$[ "tv": [ ],
		   "dvb": [ ]
		]
	];

    map READ_det =
	$[ "probe": $[ "tv": [
	  $[
		"sub_device": "jmeno prvni",
		"unique_key": "klic",
		"drivers":[ $[ "modules" : [] ] ]
	  ],
	  $[
		"device":"jmeno druhe",
		"unique_key": "123xyz",
		"drivers":[ $[ "modules": [ [ "bttv" ] ] ] ]
	  ]],
	  "dvb" : []
	]];


	// Read() returns nil and invokes Report::Warning
	/*
    TEST (``(Tv::Detect()), [ READ_nil, $[], $[] ], nil);
	*/

	// Read() returns an empty list
	TEST (``(Tv::Detect()), [ READ_empty, $[], $[] ], nil);
	DUMP (Tv::detected_cards);

	TEST (``(Tv::Detect()), [ READ_det, $[], $[] ], nil);
	DUMP (Tv::detected_cards);

}
