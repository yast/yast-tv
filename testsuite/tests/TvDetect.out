Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp"
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system nil
Read	.probe.tv []
Return	true
Dump	[]
Read	.probe.tv [$["drivers":[$["modules":[]]], "sub_device":"jmeno prvni", "unique_key":"klic"], $["device":"jmeno druhe", "drivers":[$["modules":[["bttv"]]]], "unique_key":"123xyz"]]
Return	true
Dump	[$["name":"jmeno prvni", "unique_key":"klic"], $["module":"bttv", "name":"jmeno druhe", "unique_key":"123xyz"]]
